packages/x-date-pickers-pro/src/SingleInputDateTimeRangeField/SingleInputDateTimeRangeField.types.ts

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { SlotComponentProps } from '@mui/base/utils';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import {
+  UseDateTimeRangeFieldDefaultizedProps,
+  UseDateTimeRangeFieldProps,
+} from '../internal/models';
+
+export interface UseSingleInputDateTimeRangeFieldParams<TDate, TChildProps extends {}> {
+  props: UseSingleInputDateTimeRangeFieldComponentProps<TDate, TChildProps>;
+  inputRef?: React.Ref<HTMLInputElement>;
+}
+
+export interface UseSingleInputDateTimeRangeFieldProps<TDate>
+  extends UseDateTimeRangeFieldProps<TDate> {}
+
+export type UseSingleInputDateTimeRangeFieldDefaultizedProps<
+  TDate,
+  AdditionalProps extends {},
+> = UseDateTimeRangeFieldDefaultizedProps<TDate> & AdditionalProps;
+
+export type UseSingleInputDateTimeRangeFieldComponentProps<TDate, TChildProps extends {}> = Omit<
+  TChildProps,
+  keyof UseSingleInputDateTimeRangeFieldProps<TDate>
+> &
+  UseSingleInputDateTimeRangeFieldProps<TDate>;
+
+export interface SingleInputDateTimeRangeFieldProps<TDate>
+  extends UseSingleInputDateTimeRangeFieldComponentProps<TDate, TextFieldProps> {
```

Comment:
You'll also need to import this type, but it will limit the possible props to a sub-set of props, which make sense/ 😉 
```suggestion
  extends UseSingleInputDateTimeRangeFieldComponentProps<TDate, FieldsTextFieldProps> {
```